lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -6908,33 +6943,7 @@ namespace Js
             }
         }
 
-        // handle inlined segment
-        SparseArraySegmentBase* inlineHeadSegment = nullptr;
-        bool hasInlineSegment = false;
-        // The following if else set is used to determine whether a shallow or hard copy is needed
-        if (JavascriptNativeArray::Is(pArr))
-        {
-            if (JavascriptNativeFloatArray::Is(pArr))
-            {
-                inlineHeadSegment = DetermineInlineHeadSegmentPointer<JavascriptNativeFloatArray, 0, true>((JavascriptNativeFloatArray*)pArr);
-            }
-            else if (JavascriptNativeIntArray::Is(pArr))
-            {
-                inlineHeadSegment = DetermineInlineHeadSegmentPointer<JavascriptNativeIntArray, 0, true>((JavascriptNativeIntArray*)pArr);
-            }
-            Assert(inlineHeadSegment);
-            hasInlineSegment = (startSeg == (SparseArraySegment<T>*)inlineHeadSegment);
-        }
-        else
-        {
-            // This will result in false positives. It is used because DetermineInlineHeadSegmentPointer
-            // does not handle Arrays that change type e.g. from JavascriptNativeIntArray to JavascriptArray
-            // This conversion in particular is problematic because JavascriptNativeIntArray is larger than JavascriptArray
-            // so the returned head segment ptr never equals pArr->head. So we will default to using this and deal with
-            // false positives. It is better than always doing a hard copy.
-            hasInlineSegment = HasInlineHeadSegment(pArr->head->length);
-        }
-
+        bool hasInlineSegment = IsInlineSegment(startSeg, pArr);
```

Comment:
Added nullptr checking in JavascriptArray::IsInlineSegment().

---
In reply to: [123654814](https://github.com/Microsoft/ChakraCore/pull/3211#discussion_r123654814) [](ancestors = 123654814)